iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/algorithm.inl

```diff
@@ -81,46 +82,47 @@ template <typename T, typename... ValueList>
 inline constexpr bool
 doesContainValue(const T value, const T firstValueListEntry, const ValueList... remainingValueListEntries) noexcept
 {
+    // AXIVION Next Line AutosarC++19_03-M6.2.2 : intentional check for exact equality
```

Comment:
You could add an doxygen comment, that this method probably does not make sense for float or even restrict it only to be used for non-float types using a type trait/static_assert. See [here](https://rules.sonarsource.com/c/RSPEC-1244) for an example.

Reply:
Unfortunately I can't restrict it because we use it with floating-point types and there it makes sense. If you're interested, `doesContainValue` is for instance called in `PosixCallVerificator::failureReturnValue` which in turn is called with `(-)HUGE_VAL(F)` in the `convert::fromString` specializations. Since the exact return value from the posix call is stored and we search for the exact failure return value, this is equality check is safe.

Reply:
@FerdinandSpitzschnueffler Ok, got it. We are only compare those special floats and hence can count on an exact match.

@MatthiasKillat Well, I partly disagree here. This MISRA rule `Floating-point expressions shall not be directly or indirectly tested for equality or inequality` assumes some developers might not be aware that it doesn't make much sense to check equality/ inequality with `floats`. I think it could make sense to either

1) Document that this method should be only used with care for `float`
2) Rename it e.g. `s/doesContainValue/doesContainExactValue`

Reply:
@mossmaurice Just to clarify. The equality check does not only work with those special floats like `HUGE_VAL`. If one stores 4.4, they will find 4.4 in the comparison. But if one stores a floating-point value resulting from a calculation, like 4*1.1, the comparison with 4.4 may return false. But this is the nature of floating-point operations. Regarding your suggestions: 1) I think it's now documented sufficiently in the header file. 2) Renaming the function is IMO unnecessary. Reading the current function name, I would expect an exact match and not a match "within a range". In the latter case, I would expect a different function name instead. Additionally, if developers are not aware of floating-point comparison behavior then a renaming would give them no hint either.